components/RichBody/index.tsx

```diff
@@ -190,39 +204,67 @@ export default function RichBody(props: Props): ReactElement {
       ),
       [BLOCKS.OL_LIST]: (node, children: any) => {
         return (
-          <ol dir={direction(children)} className="pb-5 ml-10 list-decimal">
+          <ol
+            dir={
+              typeof children[0] === 'object'
+                ? getTextDirectionFromNode(children[0])
+                : getDirection(children)
+            }
+            className="pb-5 ml-10 list-decimal"
+          >
             {children}
           </ol>
         );
       },
       [BLOCKS.UL_LIST]: (node, children: any) => {
         return (
-          <ul dir={direction(children)} className="pb-5 ml-10 list-disc">
+          <ul
+            dir={
+              typeof children[0] === 'object'
+                ? getTextDirectionFromNode(children[0])
+                : getDirection(children)
+            }
+            className="pb-5 ml-10 list-disc"
+          >
             {children}
           </ul>
         );
       },
-      [BLOCKS.LIST_ITEM]: (node, children) => {
+      [BLOCKS.LIST_ITEM]: (node, children: any) => {
         const renderChildren = Children.map(children, (child: any) => {
           if (child.type === 'p') {
             const newProps = {
               ...child.props,
               className: 'leading-relaxed pb-1',
             };
+
             return cloneElement(child, newProps);
           }
         });
-        return <li>{renderChildren}</li>;
+
+        return (
+          <li
+            dir={
+              typeof children[0] === 'object'
```

Comment:
This code is duplicated on line 137. Any chance we can make it a function?

Reply:
ok sure! created a new function for it. But changed the other functions name in the way, I tried to find a way to make it as declarative as possible. the "getTextDirectionFromNode" from before that was the recursive one, is now called "recursiveTextDirectionFromNode", and the new one you asked for is called  "getTextDirectionFromNode". So in a few cases I call these function and if neccesary these function will call the recursive one inside.

Reply:
Ok nice. Appreciate the clear naming!